serverless/assets/verify_expiry.private.js

```diff
@@ -0,0 +1,15 @@
+exports.handler = function (context, event, callback) {
```

Comment:
Nice, I see what you're doing here. It seems like this is acting like a middleware to your endpoints. However, the implementation requires duplicating the code block below to all your endpoints, which is not ideal.

```js
  const verifyExpiry = require(Runtime.getAssets()['/verify_expiry.js'].path);
  verifyExpiry.handler(context, event, callback);
```

How about we do this? Update this file to act like a middleware:
```js
const authMiddleware = function(handler) {
  return function(context, event, callback) {
    if (Date.now() > context.APP_EXPIRY) {
      const response = new Twilio.Response();
      response.appendHeader('Content-Type', 'application/json');
      response.setStatusCode(401);
      response.setBody({
        error: {
          message: 'passcode expired',
          explanation:
            'foo bar.',
        },
      });
      return callback(null, response);
    }
    handler(context, event, callback);
  };
};
```

Then in your endpoints handlers, for example `serverless/functions/app/turn-credentials.js`, you can use it like so:

```js
exports.handler = authMiddleware(function(context, event, callback) {
  const client = context.getTwilioClient();
  client.tokens.create({ ttl: 30 }).then((token) => callback(null, token));
});
```

Reply:
I see what you're saying.  We would still need to require() the `authMiddleware` function in each file with:

`const authMiddleware = require(Runtime.getAssets()['/auth_middleware.js'].path);`

So this will only save us one line of code per function. On one hand, I don't think it is worth doing for only two functions (saving two lines of code), but on the other hand, it is a little cleaner. I'll think about this one...

Reply:
Ok no worries. I don't have a strong opinion on this because we only have two endpoints at the moment. But it's good to have something cleaner if we end up building more on top of this.

Reply:
I think I'll keep it as is, but this `authMiddleware` function is a good pattern to keep in mind.